File: core/src/main/java/me/devsnox/planetsystem/core/api/data/PlayerData.java
package me.devsnox.planetsystem.core.api.data;

import me.devsnox.planetsystem.api.planet.LoadedPlanet;
import me.devsnox.planetsystem.api.planet.Planet;
import me.devsnox.planetsystem.api.player.PlanetPlayer;
import org.jetbrains.annotations.Nullable;

import java.util.UUID;
import java.util.function.Consumer;

public interface PlayerData {

    void load(UUID uuid, Consumer<PlanetPlayer> request);

    void save(UUID uuid);

    @Nullable
    PlanetPlayer getPlayer(UUID uuid);

    boolean isPlanetLoaded(UUID owner);

    Planet getPlanet(UUID owner);

    LoadedPlanet getLoadedPlanet(UUID owner);

}

File: api/src/main/kotlin/de/astride/planetsystem/api/holder/data/PlayerData.kt
package de.astride.planetsystem.api.holder.data

import de.astride.planetsystem.api.inline.Owner
import de.astride.planetsystem.api.player.PlanetPlayer

interface PlayerData {

    val players: Set<PlanetPlayer>

    fun load(owner: Owner, request: (PlanetPlayer) -> Unit)

    fun save(owner: Owner)

    fun unload(owner: Owner)

}

fun PlayerData.find(owner: Owner): PlanetPlayer? = players.find { it.owner == owner }
